test/Chainweb/Test/Utils.hs

```diff
@@ -647,20 +657,67 @@ withTestAppServer tls v appIO envIO userFunc = bracket start stop go
     go (_, _, env) = userFunc env
 
 
+data ValidationException = ValidationException
+    { vReq :: W.Request
+    , vResp :: (ResponseHeaders, Status, BL.ByteString)
+    , vErr :: WV.TopLevelError
+    }
+    deriving (Show, Typeable)
+
+instance Exception ValidationException
+
+{-# NOINLINE chainwebOpenApiSpec #-}
+chainwebOpenApiSpec :: OpenApi
+chainwebOpenApiSpec = unsafePerformIO $ do
```

Comment:
It is actually helping us a lot, because it is never evaluated more than once, despite the underneath `evaluate` being executed more than once. Without this, we quickly ran into github's rate limiting.

Reply:
Right! For some reason, I got the impression that the `withChainwebTestServer` below was something we were calling only once as part of the test suite setup, but I now see that's wrong.

Reply:
+1 to the use of unsafePerformIO with NOINLINE here